frameworks/MLNet/exec.py

```diff
@@ -0,0 +1,116 @@
+    with Timer() as training:
+        cmd = '{} {}'.format(mlnet, sub_command)
+
+        # dataset & test dataset
+        cmd += ' --dataset {} --test-dataset {}'.format(train_dataset, test_dataset)
+
+        # train time
+        cmd += ' --train-time {}'.format(train_time_in_seconds)
+
+        # label
+        cmd += ' --label-col label'
+
+        # output folder & name
+        cmd += ' --output {} --name {}'.format(config.output_dir, name)
+
+        # log level & log file place
+        cmd += ' --verbosity q --log-file-path {}'.format(log_path)
+        run_cmd(cmd, _env_=os.environ)
+
+        train_result_json = os.path.join(temp_output_folder, '{}.mbconfig'.format(name))
+        if not os.path.exists(train_result_json):
+            raise NoResultError("MLNet failed producing any prediction.")
+        
+    with open(train_result_json, 'r') as f:
+        json_str = f.read()
+        mb_config = json.loads(json_str)
+        model_path = mb_config['Artifact']['MLNetModelPath']
+        output_prediction_txt = config.output_predictions_file.replace('.csv', '.txt')
+        # predict
+        if config.type == 'classification':
+            predict_cmd = '{} {}'.format(mlnet, 'predict')
+            predict_cmd += ' --model {} --dataset {} --task-type classification'.format(model_path, test_dataset)
+            predict_cmd += ' > {}'.format(output_prediction_txt)
+            with Timer() as prediction:
+                run_cmd(predict_cmd)
+            prediction_df = pd.read_csv(output_prediction_txt, dtype={'PredictedLabel':'object'})
+            #rename_df = prediction_df.rename(columns={'PredictedLabel':'predictions'})
+            #rename_df['truth'] = dataset.test.y
+            #rename_df.to_csv(config.output_predictions_file)
+            save_predictions(
+                dataset=dataset,
+                output_file=config.output_predictions_file,
+                predictions=prediction_df['PredictedLabel'].values,
+                truth=dataset.test.y,
+                probabilities=prediction_df.values[:,:-1],
+                probabilities_labels=list(prediction_df.columns.values[:-1]),
+            )
+
+            return dict(
+                training_duration=training.duration,
+                predict_duration=prediction.duration,
+            )
+        
+        if config.type == 'regression':
+            predict_cmd = '{} {}'.format(mlnet, 'predict')
+            predict_cmd += ' --model {} --dataset {} --task-type regression'.format(model_path, test_dataset)
+            predict_cmd += ' > {}'.format(output_prediction_txt)
+            with Timer() as prediction:
+                run_cmd(predict_cmd)
+            prediction_df = pd.read_csv(output_prediction_txt)
+            rename_df = prediction_df.rename(columns={'Score':'predictions'})
+            rename_df['truth'] = dataset.test.y
+            rename_df.to_csv(config.output_predictions_file)
```

Comment:
can't you use `save_predictions` like you did for classification?
This guarantees better compatibility for various kind of datasets.

Reply:
### Resolved